core/instance_events.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2016-2019 Cristian Măgherușan-Stanciu
+// Licensed under the Open Software License version 3.0
+
+package autospotting
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+// returns the InstanceID, State or an error
+func parseEventData(event events.CloudWatchEvent) (string, *string, *string, error) {
+	var detailData instanceData
+	if err := json.Unmarshal(event.Detail, &detailData); err != nil {
+		logger.Println(err.Error())
+		return "", nil, nil, err
+	}
+	eventType := event.DetailType
+
+	// Amazon EC2 State Change Events
+	if eventType == "EC2 Instance State-change Notification" &&
+		detailData.InstanceID != nil &&
+		detailData.State != nil {
+		return "IS", detailData.InstanceID, detailData.State, nil
```

Comment:
Looks much better, I would just like the "IS" and other strings including those of EventBridge events to be defined as constants with descriptive names.

Reply:
can you give me an example, please

Reply:
something like this
```
const InstanceStateNotficationMessage = "EC2 Instance State-change Notification"
const InstanceStateChange = "ISC" // probably better named than "IS"
```
and so on...

The benefit is we get IDE support for them which reduces the chance of errors and increases clarity of the code.